File: app/src/main/java/com/bracketcove/postrainer/PostrainerApplication.java
package com.bracketcove.postrainer;

import android.app.Application;

import com.bracketcove.postrainer.dependencyinjection.components.ApplicationComponent;
import com.bracketcove.postrainer.dependencyinjection.components.DaggerApplicationComponent;
import com.bracketcove.postrainer.dependencyinjection.modules.ApplicationModule;
import com.squareup.leakcanary.LeakCanary;

import io.realm.Realm;
import io.realm.RealmConfiguration;


public class PostrainerApplication extends Application {
    private ApplicationComponent applicationComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        initializeRealm();
        initializeLeakCanary();

        ApplicationModule applicationModule = new ApplicationModule(
                getApplicationContext()
        );

        applicationComponent = DaggerApplicationComponent
                .builder()
                .applicationModule(applicationModule)
                .build();

    }

    private void initializeRealm(){
        Realm.init(this);
        RealmConfiguration realmConfiguration = new RealmConfiguration.Builder().build();
        Realm.setDefaultConfiguration(realmConfiguration);
    }

    private void initializeLeakCanary(){
        if (LeakCanary.isInAnalyzerProcess(this)) {
            return;
        }
        LeakCanary.install(this);
    }

    public ApplicationComponent getApplicationComponent() {
        return this.applicationComponent;
    }
}

File: app/src/main/java/com/bracketcove/postrainer/PostrainerApplication.kt
package com.bracketcove.postrainer

import android.app.Application
import com.bracketcove.postrainer.dependencyinjection.AndroidReminderProvider
import io.realm.Realm
import io.realm.RealmConfiguration


class PostrainerApplication: Application() {

    internal val dependencyProvider = AndroidReminderProvider(this)

    override fun onCreate() {
        super.onCreate()
        initializeRealm()



    }



    private fun initializeRealm() {
        Realm.init(this)
        val realmConfiguration = RealmConfiguration.Builder()
            .build()
        Realm.setDefaultConfiguration(realmConfiguration)
    }


}
